aiida/orm/nodes/process/calculation/calcjob.py

```diff
@@ -242,11 +244,26 @@ def set_state(self, state: CalcJobState) -> None:
 
         :raise: ValueError if state is invalid
         """
+        from aiida.manage.configuration import get_config_option
+
         if not isinstance(state, CalcJobState):
             raise ValueError(f'{state} is not a valid CalcJobState')
 
         self.set_attribute(self.CALC_JOB_STATE_KEY, state.value)
 
+        if get_config_option('calcjob_state_tracking'):
+            # it is used like this in a method below, any difference with what I am doing?
+            #timestamp = timezone.datetime_to_isoformat(timezone.now())
+            timestamp = datetime.datetime.utcnow().replace(tzinfo=datetime.timezone.utc).isoformat()
+
+            state_tracker = self.get_attribute(self.CALC_JOB_STATE_TRACKER, None)
+            if state_tracker is None:
+                state_tracker = dict()
+
+            state_tracker[str(timestamp)] = state.value
```

Comment:
I'm a bit puzzled by this choice of key and value: I think it makes more sense to invert? I.e. the key is the state.value, and the value is the timestamp? I doubt you want to discover, given the exact timestamp, the state corresponding to it. You will most probably want to know the time, given the state change you're interested into.

Reply:
I did it this way because I was not sure what happens when you pause/play calcjobs, or when you restart the daemon in the middle of the run. Is there no risk ever of the daemon trying to set again previously set states and overwriting the keys? I thought having the timestamp as the key ensures it will be unique and also provides a natural order for the information.

Reply:
It is possible (maybe?) but probably ok to keep only the latest time of change (I would use this only for monitoring in the case everything is running at max speed, no daemon stops, no pauses etc.
Actually, what could be "worse" is if two different processes read the dictionary, modify and rewrite it at the same time. But this should not happen as only one daemon worker should act on a given calcjob node at any given time.

Reply:
Ok, I still kind of think that using timestamp as the key is the more robust way to do it (even if it makes it a bit more uncomfortable to work with later), but maybe we can review this when incorporating the changes to aiida-core. Now for this LUMI test I will just switch them as you said and we'll work with that.